content/automate/ManageIQ/Transformation/Infrastructure/VM/vmwarews.class/__methods__/utils.rb

```diff
@@ -7,53 +7,44 @@ module VMware
             class Utils
               require 'rbvmomi'
 
-              def initialize(handle = $evm)
-                @handle = handle
+              def self.host_fingerprint(host, handle = $evm)
+                require 'socket'
+                require 'openssl'
+
+                tcp_client = TCPSocket.new(host.ipaddress, 443)
+                ssl_context = OpenSSL::SSL::SSLContext('SSLv23_client')
+                ssl_content.verify_mode = OpenSSL::SSL::VERIFY_NONE
+                ssl_client = OpenSSL::SSL::SSLSocker.new(tcp_client, ssl_context)
+                cert = OpenSSL::X509::Certificate.new(ssl_client.peer_cert)
+                ssl_client.sysclose
+                tcp_client.close
+
+                Digest::SHA1.hexdigest(cert.to_der).upcase.scan(/../).join(":")
               end
 
-              def main
+              def self.provider_connection(ems, handle = $evm)
+                @provider_connection ||= RbVmomi::VIM.connect(:host => ems.ipaddress || ems.hostname, :user => ems.authentication_userid, :password => ems.authentication_password, :insecure => true)
```

Comment:
@fdupont-redhat should the secure true/false be configurable. This makes the assumption its always insecure

Reply:
@fdupont-redhat why is the provider_connection an instance variable, this looks like a class method. This connection cannot have scope outside of the process that is using this, once the process ends the provider_connection is meaningless. Also we need to ensure that this provider connection doesn't get into a $evm.set_state_var.
So could the provider connection not be exposed to callers maybe we could pass in a block into this utility method and it would execute the block on that provider. Or have a shared class that actually has instance methods which can execute the code on the provider but keeps the connection private and doesn't allow other classes access to it.
Exposing the connection object this way can lead to misuse.

Reply:
@mkanoor To be honest, I would remove these methods: `provider_connection`, `vm_get_ref`, `vm_rename`. I think they should be in the model and only `vm_rename` should be exposed in `vm` class. This is currently unused as we don't rename the VMs after migration. I just kept them because it was already there.